JS/app.js

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable max-classes-per-file */
+class Book {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
+}
+
+class Store {
+  static getBooks() {
+    let books;
+    if (localStorage.getItem('books') === null) {
+      books = [];
+    } else {
+      books = JSON.parse(localStorage.getItem('books'));
+    }
+
+    return books;
+  }
+
+  static addBook(book) {
+    const books = Store.getBooks();
+    books.push(book);
+    localStorage.setItem('books', JSON.stringify(books));
+  }
+
+  static removeBook(author) {
+    const books = Store.getBooks();
+
+    books.forEach((book, index) => {
+      if (book.author === author) {
+        books.splice(index, 1);
+      }
+    });
+
+    localStorage.setItem('books', JSON.stringify(books));
+  }
+}
+
+class showBooks {
+  static displayBooks() {
+    const books = Store.getBooks();
+
+    books.forEach((book) => showBooks.addBookToList(book));
+  }
+
+  static addBookToList(book) {
+    const list = document.querySelector('#list');
+    const row = document.createElement('tr');
+
+    row.innerHTML = `
+        <td>${book.title}</td>
+        <td> by ${book.author}</td>
+        <td><button class="delete">Remove book</button></td>
+      `;
+
+    list.appendChild(row);
+  }
+
+  static deleteBook(el) {
+    if (el.classList.contains('delete')) {
+      el.parentElement.parentElement.remove();
+    }
+  }
+
+  static clearFields() {
+    document.querySelector('#title').value = '';
+    document.querySelector('#author').value = '';
+  }
+}
+
+document.addEventListener('DOMContentLoaded', showBooks.displayBooks);
+
+document.querySelector('#form').addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const title = document.querySelector('#title').value;
+  const author = document.querySelector('#author').value;
+
+  const book = new Book(title, author);
+
+  showBooks.addBookToList(book);
+  Store.addBook(book);
+  showBooks.clearFields();
+});
+
+document.querySelector('#list').addEventListener('click', (e) => {
+  showBooks.deleteBook(e.target);
+
+  Store.removeBook(e.target.parentElement.previousElementSibling.textContent);
+});
```

Comment:
- [ ] The remove books functionality does not seem to be working on `localStorage` level.

Look I add one book:

![image](https://user-images.githubusercontent.com/57421823/153093673-39b038ef-012e-4916-8f62-8675c5598f40.png)

Then, I delete it, putting a console log on this event listener:

![image](https://user-images.githubusercontent.com/57421823/153093752-55750849-d1b2-4871-ba6e-6bb26a3eb165.png)

After that, I try to remove the book:

![image](https://user-images.githubusercontent.com/57421823/153093829-2d4b5fad-d32f-454a-9a85-365ba1920157.png)

It removes the book from the DOM, but not from localStorage.

Therefore, it is not working because it takes the author as "by test author" instead of just "test author".  That's why since it does not find any matching author, it deletes nothing from local storage.

You could:

- Use regex so the "by" is ignored from the search or...
- Make the "by" its own element, so it does not interfere with the deletion searches.

That way, upon searching, the function will find the correct matching author and delete the element.

I hope this helps 💪 .